packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +211,134 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The service Role of this Deployment Group.
+   */
+  public readonly role: iam.IRole;
+
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    for (const service of props.services) {
+      if (service instanceof ecs.BaseService) {
```

Comment:
I think instead of `instanceof` you can do

```ts
if (Resource.isOwnedResource(service))
```

Reply:
This one manipulates a property defined in BaseService, so it needs to check for that class

Reply:
But the input property accepts `IBaseService` so the value is either a `BaseService` (owned) or
`IBaseService` (imported). It looks like we don't have this documented anywhere, but `instanceof`
can be [problematic](https://github.com/aws/constructs/pull/955) and we are working on removing instances of it.

Reply:
got it, done